File: apps/codemap-search/experiments/event-navigation-poc/language-corpus/regressions/examples/dart/controls/boundaries.dart
class Router {
  (void Function(), void Function()) callbacks = (() {}, () {});
  void install(void Function() f, void Function() g) {
    callbacks = (f, g); // @S1
  }
  void first() {
    final (callback, _) = callbacks;
    callback(); // @I1
  }
  void second() {
    final (_, callback) = callbacks;
    callback(); // @I2
  }
}

File: apps/codemap-search/experiments/event-navigation-poc/language-corpus/regressions/examples/scala/controls/boundaries.scala
class Router {
  var callbacks: (() => Unit, () => Unit) = (() => (), () => ())
  def install(f: () => Unit, g: () => Unit): Unit = {
    callbacks = (f, g) // @S1
  }
  def first(): Unit = {
    val (callback, _) = callbacks
    callback() // @I1
  }
  def second(): Unit = {
    val (_, callback) = callbacks
    callback() // @I2
  }
}
